packages/astro/src/cli/add.ts

```diff
@@ -0,0 +1,361 @@
+	}
+
+	const integrations = await validateIntegrations(names);
+
+	// Add integrations to astro config
+	// TODO: At the moment, nearly nothing throws an error. We need more errors!
+	let ast: t.File | null = null;
+	try {
+		ast = await parseAstroConfig(configURL);
+
+		debug('add', 'Parsed astro config');
+
+		const defineConfig = t.identifier('defineConfig');
+		ensureImport(ast, t.importDeclaration([t.importSpecifier(defineConfig, defineConfig)], t.stringLiteral('astro/config')));
+		wrapDefaultExport(ast, defineConfig);
+
+		debug('add', 'Astro config ensured `defineConfig`');
+
+		for (const integration of integrations) {
+			await addIntegration(ast, integration);
+			debug('add', `Astro config added integration ${integration.id}`);
+		}
+	} catch (err) {
+		debug('add', 'Error parsing/modifying astro config: ', err);
+		info(
+			logging,
+			null,
+			"Sorry, we couldn't update your configuration automatically. [INSERT HOW TO DO IT MANUALLY --- this link might help: https://next--astro-docs-2.netlify.app/en/guides/integrations-guide/]"
+		);
+	}
+
+	let configResult: UpdateResult | undefined;
+	let installResult: UpdateResult | undefined;
+
+	if (ast) {
+		try {
+			configResult = await updateAstroConfig({ configURL, ast, logging });
+		} catch (err) {
+			debug('add', 'Error updating astro config', err);
+			error(logging, null, 'There has been an error updating the astro config. You might need to update it manually.');
+			return;
+		}
+	}
+
+	switch (configResult) {
+		case UpdateResult.cancelled: {
+			info(logging, null, msg.cancelled(`Your configuration has not been updated.`));
+			return;
+		}
+		case UpdateResult.none: {
+			info(logging, null, msg.success(`Configuration up-to-date. No changes needed!`));
+			break;
+		}
+	}
+
+	installResult = await tryToInstallIntegrations({ integrations, cwd, logging });
+
+	switch (installResult) {
+		case UpdateResult.updated: {
+			const len = integrations.length;
+			if (integrations.find((integration) => integration.id === 'tailwind')) {
+				const DEFAULT_TAILWIND_CONFIG = `module.exports = {
+	content: [],
+	theme: {
+		extend: {},
+	},
+	plugins: [],
+}\n`;
+				await fs.writeFile(fileURLToPath(new URL('./tailwind.config.mjs', configURL)), DEFAULT_TAILWIND_CONFIG);
+			}
+			info(logging, null, msg.success(`Added ${len} integration${len === 1 ? '' : 's'} to your project`));
+			return;
+		}
+		case UpdateResult.cancelled: {
+			info(logging, null, msg.cancelled(`No dependencies installed.`, `Be sure to install them manually before continuing!`));
+			return;
+		}
+		case UpdateResult.failure: {
+			info(logging, null, msg.failure(`There was a problem installing dependencies.`, `Be sure to install them manually before continuing!`));
+			process.exit(1);
+		}
+	}
+}
+
+async function parseAstroConfig(configURL: URL): Promise<t.File> {
+	const source = await fs.readFile(fileURLToPath(configURL), { encoding: 'utf-8' });
```

Comment:
do we still support the "astro" config in your package.json through proload?

also, I'm assuming `astro add` doesn't support `astro.config.ts` since it would fail these parse steps. I think that's a fine limitation for now as long as we handle it gracefully.

Reply:
I'm adding a specific state for `package.json#astro`. And `astro.config.ts` does work! Babel can handle it.

Reply:
Nice! I feel like we might want to drop the `package.json` support but I'm happy to push that convo to another time